### ISSUE TITLE
Prow test are failing due to nodejs-8 image

### ISSUE THREAD

**Issue by kadel** [openshift/odo#2641]
Prow test are failing due to nodejs-8 image!
 

ci/prow/v4.2-integration-e2e-benchmark 
ci/prow/v4.3-integration-e2e-benchmark 

```
• Failure [397.878 seconds]
odo supported images e2e tests
/go/src/github.com/openshift/odo/tests/e2escenarios/e2e_images_test.go:14
  odo supported images deployment
  /go/src/github.com/openshift/odo/tests/e2escenarios/e2e_images_test.go:137
    Should be able to verify the nodejs-8-centos7 image [It]
    /go/src/github.com/openshift/odo/tests/e2escenarios/e2e_images_test.go:173
    Timeout out after � minutes
    /go/src/github.com/openshift/odo/tests/helper/helper_generic.go:122
------------------------------
```

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2637/pull-ci-openshift-odo-master-v4.2-integration-e2e-benchmark/684

/priority critical
/area testing
/kind failing-test

**Comment by amitkrout** [comment]
@prietyc123 @kadel I have tested the same scenario on 4.2 cluster more than 5 times to have atleast one occurrence of failure but none of the the time it hit. 

From the logs it is clear that curl command could not hit the route in the specified time limit. Not sure if this is some kind of glitch from network side or from the nodejs application side itself. @kadel Until we are not sure on the issue behaviour till then can we add flake in the title.

**Comment by kadel** [comment]
Yesterday I saw almost all PRs failing on this.
/kind flake
/priority high
/remove-priority critical
/remove-kind failing-test

**Comment by kadel** [comment]
This is getting really really annoying :-/ 

Another occurrence in 

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2629/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/754

this time for centos7-s2i-nodejs:10.x image

**Comment by prietyc123** [comment]
@amitkrout I am able to reproduce the error with ```nodejs-10:latest``` and ```centos7-s2i-nodejs:10.x``` images on 4.2 cluster. 

Curl command could not be able to hit for both the images. Need to figure out the reason behind this.

**Comment by prietyc123** [comment]
@kadel I ran the test yesterday and today also in a loops but now I am not getting this error. Although I will be running it again and will update further.

**Comment by amitkrout** [comment]
The same failure first time hit in java:8 component too against 4.1 cluster - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2755/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/1301#1:build-log.txt%3A885

Tried locally twice and none of the time i hit the issue. Now i am going to send a pr to get atleast few info like ```oc get pod, oc get dc, oc get dc -o yaml``` just after the component push may be helpful to debug the issue.

**Comment by amitkrout** [comment]
Ahh...
Finally got the problem. I think the issue is with watch function pointing to another context dir (Due to the race condition in parallel run).

Will send the PR today itself

**Comment by prietyc123** [comment]
Still I can observe the same issue on prow : https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2809/pull-ci-openshift-odo-master-v4.2-integration-e2e-benchmark/1857#1:build-log.txt%3A722

**Comment by amitkrout** [comment]
> Still I can observe the same issue on prow : https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2809/pull-ci-openshift-odo-master-v4.2-integration-e2e-benchmark/1857#1:build-log.txt%3A722

@prietyc123 Thanks for reporting it. Will take a look.

**Comment by prietyc123** [comment]
one more occurance : https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/3041/pull-ci-openshift-odo-master-v4.3-integration-e2e/72#1:build-log.txt%3A732

**Comment by prietyc123** [comment]
hitting the same failure again https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/3041/pull-ci-openshift-odo-master-v4.2-integration-e2e/75#1:build-log.txt%3A748

**Comment by dharmit** [comment]
Seeing a failure for this on PR #3239:

* 4.2 integration tests failure for `centos7-s2i-nodejs` image - [link to logs](https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/3239/pull-ci-openshift-odo-master-v4.2-integration-e2e/676)
* 4.5 integration tests failure for `nodejs-8-rhel7` image - [link to logs](https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/3239/pull-ci-openshift-odo-master-v4.5-integration-e2e/50).

**Comment by girishramnani** [comment]
we would remove the nodejs 8 image related tests

**Comment by girishramnani** [comment]
@amitkrout please open a separate issue for more of the scope discussed

**Comment by amitkrout** [comment]
> @amitkrout please open a separate issue for more of the scope discussed

@girishramnani As discussed i have created separate issue https://github.com/openshift/odo/issues/3307. 

Also we had a discussion on the supported images that nodejs8 has been deprecated, @girishramnani however i can still see these images in the supported list https://github.com/openshift/odo-init-image/blob/master/language-scripts/image-mappings.json#L5-L15 and in the odo README too - https://github.com/openshift/odo#officially-supported-languages-and-corresponding-container-images. We should remove those reference aswell, right ?

**Comment by dharmit** [comment]
@amitkrout to do write-up explaining the problem and how to fix it.

**Comment by amitkrout** [comment]
> @amitkrout to do write-up explaining the problem and how to fix it.

I have created a separate issue #3307 as per the comment https://github.com/openshift/odo/issues/2641#issuecomment-638865321. The cause of failure and fix will be discussed in #3307.

Actually i am waiting for @girishramnani comment on my last comment. Then only i can decide the next task, otherwise i will close the issue.

**Comment by girishramnani** [comment]
why do we still have nodejs-8 images?
